pkg/azure/provider.go

```diff
@@ -249,6 +254,28 @@ func (p *Provider) GetServicePrincipalToken(env *azure.Environment, resource str
 		err = fmt.Errorf("nmi response failed with status code: %d", resp.StatusCode)
 		return nil, err
 	}
+
+	if p.UseVMManagedIdentity {
+		msiEndpoint, err := adal.GetMSIVMEndpoint()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get managed identity (MSI) endpoint")
+		}
+
+		if p.UserAssignedIdentityID != "" {
+			r, _ := regexp.Compile(`^(\S{4})(\S|\s)*(\S{4})$`)
```

Comment:
Not a blocker for this PR, but we should move the client id redaction to a util function and reuse everywhere

Reply:
Done!